### ISSUE TITLE
Error `collection with delete-orphan` When I try migrated to version 24.0.2

### ISSUE THREAD

**Issue by jorozcof** [keycloak/keycloak#28338]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/client-java

### Describe the bug

I am migrating Keycloak 20.0.3 to Keycloak 24.0.2. The migratation it was succesfull, but When I try get all users, I have got the following error:

![image](https://github.com/keycloak/keycloak/assets/115189644/a849157b-f4e2-435a-b166-1a24a9bab744)

I am using a dependecy `keycloak-admin-client` whit spring boot.
I have got an error when I use the following: `return keycloakRealm.users().get(id).toRepresentation();`
Also a important point, the first time that I make the request, I have got the error for the first user. But on the second request I am not getting the error. After getting the error for all users, I now see all users without problems.

Also, When I restart the container, I got the same problem. Once that I got the error for all user, the response is successfull.

### Version

24.0.2

### Regression

- [ ] The issue is a regression

### Expected behavior

I hope to get all the users without any errors.

### Actual behavior

When I try get all user I have got this error por each user.

```
2024-04-01 21:16:48,622 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.keycloak.models.jpa.entities.UserEntity.requiredActions
        at org.hibernate.engine.internal.Collections.prepareCollectionForUpdate(Collections.java:271)
        at org.hibernate.engine.internal.Collections.processReachableCollection(Collections.java:231)
        at org.hibernate.event.internal.FlushVisitor.processCollection(FlushVisitor.java:54)
        at org.hibernate.event.internal.AbstractVisitor.processValue(AbstractVisitor.java:92)
        at org.hibernate.event.internal.AbstractVisitor.processValue(AbstractVisitor.java:59)
        at org.hibernate.event.internal.AbstractVisitor.processEntityPropertyValues(AbstractVisitor.java:53)
        at org.hibernate.event.internal.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:161)
        at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
        at org.hibernate.event.internal.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:226)
        at org.hibernate.event.internal.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:90)
        at org.hibernate.event.internal.DefaultAutoFlushEventListener.onAutoFlush(DefaultAutoFlushEventListener.java:52)
        at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
        at org.hibernate.internal.SessionImpl.autoFlushIfRequired(SessionImpl.java:1366)
        at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.lambda$new$3(ConcreteSqmSelectQueryPlan.java:164)
        at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.withCacheableSqmInterpretation(ConcreteSqmSelectQueryPlan.java:362)
        at org.hibernate.query.sqm.internal.ConcreteSqmSelectQueryPlan.performScroll(ConcreteSqmSelectQueryPlan.java:311)
        at org.hibernate.query.sqm.internal.QuerySqmImpl.doScroll(QuerySqmImpl.java:593)
        at org.hibernate.query.spi.AbstractSelectionQuery.scroll(AbstractSelectionQuery.java:531)
        at org.hibernate.query.spi.AbstractSelectionQuery.stream(AbstractSelectionQuery.java:548)
        at org.hibernate.query.spi.AbstractSelectionQuery.getResultStream(AbstractSelectionQuery.java:542)
        at org.keycloak.models.jpa.JpaUserProvider.getFederatedIdentitiesStream(JpaUserProvider.java:801)
        at org.keycloak.storage.UserStorageManager.getFederatedIdentitiesStream(UserStorageManager.java:747)
        at org.keycloak.models.cache.infinispan.UserCacheSession.getFederatedIdentitiesStream(UserCacheSession.java:624)

```

### How to Reproduce?

In my case, after to migrate Keycloak 20.0.3 to Keycloak 24.0.2 I have got this error when I try get users by dependency `keycloak-admin-client` in version 24.0.2

### Anything else?

I remember I tryed the migration in the version 23.0.0 and I  got the same problem.

**Comment by sguilhen** [comment]
@jorozcof Just to clarify: you also attempted to migrate 20.0.3 to 23.0.0 and got the same problem?

**Comment by jorozcof** [comment]
> @jorozcof Just to clarify: you also attempted to migrate 20.0.3 to 23.0.0 and got the same problem?

Yes I did. In fact, I did try with migrate to 21, 22 and 23. But I think that it is my database. I feel that have a wrong value or somenthing like that.
I have mounted a test database, from 20.0.3, I did some test, after that, I migrated to 23.0.0 and i didnt have any problem. 
But the werid here, is if you request the user for first time, i got the error, but if I request a second time, i didnt get the error.  Also if I restart the container or user clean cache, I have the same behavor.

**Comment by sguilhen** [comment]
This is strange. Looking at the UserResource endpoint I can't see where the user required actions reference might be changed. If it was changed when getting the user representation, for example, then searching the federated identities would likely lead to the above error, but I can't see it being changed anywhere.

@ahus1 Have you ever seen this error from hibernate or have any idea what could lead to it?

**Comment by jorozcof** [comment]
Recently I tried with keycloak 25.0.1, I could find the problem. When I requested the users, I had a method that passed the userId as a parameter and search the user in keycloak database, something like this: `keycloakRealm.users().get(userId).toRepresentation();`
But I realized that in my local database, I had some ids with UpperCase, so just what i did was, change for lowerCase, and this work it for me.

Before the version 20.0.3, I hadn't this problem.
